module/VuFind/src/VuFind/Controller/AbstractRecord.php

```diff
@@ -955,8 +957,11 @@ protected function showTab($tab, $ajax = false)
         $view->backgroundTabs = $this->getBackgroundTabs();
         $view->tabsExtraScripts = $this->getTabsExtraScripts($view->tabs);
         $view->loadInitialTabWithAjax
-            = isset($config->Site->loadInitialTabWithAjax)
-            ? (bool)$config->Site->loadInitialTabWithAjax : false;
+            = isset($config->Site->loadInitialTabWithAjax) && (bool) $config->Site->loadInitialTabWithAjax;
+        try {
+            $this->layout()->setVariable('searchOrigin', SearchOriginFactory::createObject($this->params()->fromQuery()));
```

Comment:
I think you'll want to register SearchOriginFactory as a top-level service instead of calling it statically -- otherwise VuFind's normal mechanism for extending, overriding and customizing code cannot be applied to it. Again, I can help with the details of this as needed.

Reply:
Ok, I'll look into it, I'll come back to you if I need help

Reply:
I made the changes, is it what you were expecting

Reply:
Yes, that looks good to me! Thanks!